src/Agent.Plugins/PipelineCache/PipelineCacheServer.cs

```diff
@@ -153,26 +155,106 @@ public class PipelineCacheServer
             return pipelineCacheClient;
         }
 
-        private Task DownloadPipelineCacheAsync(
+        private async Task DownloadPipelineCacheAsync(
+            AgentTaskPluginExecutionContext context,
             DedupManifestArtifactClient dedupManifestClient,
             DedupIdentifier manifestId,
             string targetDirectory,
             CancellationToken cancellationToken)
         {
 
+            
             DownloadDedupManifestArtifactOptions options = DownloadDedupManifestArtifactOptions.CreateWithManifestId(
                 manifestId,
                 targetDirectory,
                 proxyUri: null,
                 minimatchPatterns: null);
-            using( Process tarring = new Process())
+            
+            var processTcs = new TaskCompletionSource<int>();
+
+            using(var cancelSource = new CancellationTokenSource())
+            using(var linkedSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken, cancelSource.Token))
+            using(var process = new Process())
             {
-                tarring.StartInfo.FileName = "tar";
-                tarring.StartInfo.Arguments = $"-xf - -C {targetDirectory}";
-                tarring.StartInfo.UseShellExecute = false;
-                tarring.StartInfo.RedirectStandardInput = true;
-                tarring.Start();
-                return dedupManifestClient.DownloadPipelineCacheAsync(options, cancellationToken, tarring.StandardInput.BaseStream);
+                process.StartInfo.FileName = @"C:\Program Files\7-Zip\7z.exe"; // tar // @"C:\Program Files\7-Zip\7z.exe"
```

Comment:
I assume we're not going to leave the full path to 7z.exe in the code, but assume/require that `7z` is in the path?

Reply:
Correct.